### ISSUE TITLE
[loc] Loc.make_loc needs to go away.

### ISSUE THREAD

**Issue by ejgallego** [coq/coq#17031]
The `make_loc` API creates locations that lack line number information; this is a big burden for IDEs which need to handle two kinds of locations then and implement a lot of code then for client-side line number handling and caching.

Given that the usage is around 20 items (most in metasyntax), we should remove this API.

**Comment by SkySkimmer** [comment]
The make_loc occurrences in metasyntax are actually bound to https://github.com/coq/coq/blob/41e76f5c36db666637afe02bb1ca05b6ad1fb71f/vernac/metasyntax.ml#L138 ie `Loc.shift_loc` not `Loc.make_loc`

**Comment by ejgallego** [comment]
Indeed, thanks @SkySkimmer ! There just very few locations that actually use this, like 8
